src/Pages/Expenses.cshtml.cs

```diff
@@ -143,14 +151,16 @@ private async Task SetData()
             {
                 UserId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
                 var entryList = await EntryDatabase.ReadListAsync(UserId, EntryType.Expense);
-                Expenses = entryList.GetFrom(StartDate).GetTo(EndDate);
-                AllExpenses = entryList.GetSum();
+                Expenses = entryList.GetFrom(StartDate).GetTo(EndDate);//If we want to remove this I need to get total amount of these as well as their sum
+                ExpensesToDisplay = Expenses.OrderByDescending(x=> x.Date).Skip((CurrentPage - 1) * PageSize).Take(PageSize);//I think this should be done by backend?
+                AllExpenses = Expenses.GetSum();
```

Comment:
I'll try to move as much as I can to the back end, but it won't be trivial to do so it will take some time. Otherwise, this works really well and I like how you implemented pagination